task/update-infra-deployments/0.1/update-infra-deployments.yaml

```diff
@@ -87,6 +87,14 @@ spec:
       script: |
         cd cloned
         git status -s --porcelain | cut -c4- > ../updated_files.txt
+        # Check if ../updated_files.txt is empty or not
+        if [ -s ../updated_files.txt ]; then
+          echo -e "Script has modified files:\n$(cat ../updated_files.txt)"
+        else
+          echo "No modified files after running the script, exiting..."
+          exit 1
```

Comment:
When can the "No modified files after running the script" situation happen? Is it an error or can it happen in normal conditions?

Reply:
(Technically, since the script is a param, I guess normal conditions. But based on our typical usage?)

Reply:
It an error condition, like when we updated the script [here](https://github.com/redhat-appstudio/build-service/pull/239/files) since the[ infra-deployement repo PR](https://github.com/redhat-appstudio/infra-deployments/pull/2988) is yet to merge, which will have the yaml files updated by task.. so in push pipeline of build-service [PR](https://github.com/redhat-appstudio/build-service/pull/239/files), it didnot produce any changed file in `../updated_files.txt` and in result of that the following step of the task [closed the existing open PR](https://github.com/redhat-appstudio/infra-deployments/pull/2976)